### DISCUSSION TITLE
Multibody closed loop kinematic

### DISCUSSION THREAD

**Discussion by MattxFR** [OpenModelica/OpenModelica#11828] [Q&A]
Dear Open Modelica community, 

I am testing and evaluating Openmodelica performances to model 3D kinematics in general using OMEdit OpenModelica v1.22.1 (64-bit). I want to use standard Modelica library package, in particular Multibody.
While the kinematic made of joints in series seems to work very well, I have difficulties with anything where the kinematic form a loop to structure. 
I successfully modelled some 4 bar planar, and 4 bar of which one anchor moves with a prismatic joint.

However, some kinematics which are analytically solvable and rather simple, fail to compute the way I model them. 

Lets take an example of 4 bar. The first image is before the loop it is closed, i believe it proves the correct initialisation of the positions, angles and speed. The second image is after the loop is closed.
I know there is a degree of mobility remaining in the bar at the top (hyperstatism degree = -1), however this degree is needed for the second kinematic loop (non planar kinematic) in the complete mechanism I am trying to model (NB, the complete mechanism with the 2 kinematic loop also fails).

I tried spherical joint and combination of 3 revolutes, both with same failure. I also tried to use JointSSR

![image](https://github.com/OpenModelica/OpenModelica/assets/79916152/b5e5994c-ea64-4ad7-b734-ed69fd4ea910)

![image](https://github.com/OpenModelica/OpenModelica/assets/79916152/3af44f9b-71d1-4bbf-ab41-3825d80d838a)

Despite looking into many literature i could not find my errors, or if it is a limitation of Openmodellica at the moment. 
Anybody have an idea?
If its a limitation of the current capability of the solver i'm interrested to know if alternative software using Modelica already manage to handle this increased complexity of closed loop?

All the best, Matt

**Comment by casella** [comment]
@MattxFR if you post your models I can try them with Dymola. If they run, it's likely that we have an OpenModelica issue, otherwise it's likely to be a modelling issue.

In general closed loops do work in OpenModelica, you can check the [StewartPlatform](https://github.com/looms-polimi/Parallel_manipulators/tree/master/Modelica_models/StewartPlatform) model, that we [test](https://libraries.openmodelica.org/branches/master/StewartPlatform/StewartPlatform.html) regularly and successfully.

**Reply by MattxFR** [reply]
Thanks a lot Francesco, I'll read the paper and study the models associated its very interesting
